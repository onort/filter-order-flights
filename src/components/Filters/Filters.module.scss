@import "../../styles/main.scss";

.container {
  width: 100%;

  @include forNonDesktop {
    display: none;
  }
}
