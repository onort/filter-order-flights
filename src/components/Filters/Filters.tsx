/* tslint:disable no-console */
import React, { Component } from "react"

import styles from "./Filters.module.scss"
import { Card, CheckBox, FilterSection, Slider } from "../"

class Filters extends Component {
  public handleChange = () => console.log("Checkbox clicked")
  public render() {
    return (
      <Card className={styles.container}>
        <FilterSection title="Aktarma" canSelectAll={true}>
          <CheckBox
            label="Sadece Aktarmasız"
            onChange={this.handleChange}
            checked={true}
          />
          <CheckBox label="1 Aktarma" onChange={this.handleChange} />
          <CheckBox label="2+ Aktarmasız" onChange={this.handleChange} />
        </FilterSection>
        <FilterSection title="Sınıf" canSelectAll={true}>
          <CheckBox
            label="Ekonomi"
            onChange={this.handleChange}
            checked={true}
          />
          <CheckBox label="Business" onChange={this.handleChange} />
          <CheckBox label="Kurumsal" onChange={this.handleChange} />
        </FilterSection>
        <FilterSection title="Max Fiyat">
          <Slider
            min={0}
            max={100}
            onChange={this.handleChange}
            currentValue={50}
          />
        </FilterSection>
        <FilterSection title="Max Uçuş Süresi">
          <Slider
            min={0}
            max={100}
            onChange={this.handleChange}
            currentValue={50}
          />
        </FilterSection>
        <FilterSection title="Havayolu" canSelectAll={true}>
          <CheckBox
            label="Turkish Airlines"
            onChange={this.handleChange}
            checked={true}
          />
          <CheckBox label="Pegasus" onChange={this.handleChange} />
          <CheckBox label="Borajet" onChange={this.handleChange} />
        </FilterSection>
      </Card>
    )
  }
}

export default Filters
