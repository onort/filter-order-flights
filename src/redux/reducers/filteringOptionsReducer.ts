import { Reducer } from "redux"

import { DataActionTypes, FetchDataAction } from "../actions"

const filteringOptionsReducer: Reducer<{}, FetchDataAction> = (
  state = {},
  action: FetchDataAction
) => {
  switch (action.type) {
    case DataActionTypes.FETCH_DATA:
      return action.data.filteringOptions
    default:
      return state
  }
}

export default filteringOptionsReducer
