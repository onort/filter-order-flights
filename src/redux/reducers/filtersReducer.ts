import { Reducer } from "redux"

import {
  DataActionTypes,
  FetchDataAction,
  FilterActionTypes,
  SetFilterAction
} from "../actions"
import { FiltersState } from "../../types"

const initialState: FiltersState = {
  maxDuration: 0,
  maxPrice: 0
}

const filtersReducer: Reducer<
  FiltersState,
  SetFilterAction | FetchDataAction
> = (state = initialState, action: SetFilterAction | FetchDataAction) => {
  switch (action.type) {
    case DataActionTypes.FETCH_DATA:
      const { filteringOptions } = action.data
      return {
        maxDuration: filteringOptions.durationRange.max,
        maxPrice: filteringOptions.priceRange.max
      }
    case FilterActionTypes.SET_DURATION_FILTER:
      return { ...state, maxDuration: action.value }
    case FilterActionTypes.SET_PRICE_FILTER:
      return { ...state, maxPrice: action.value }
    default:
      return state
  }
}

export default filtersReducer
