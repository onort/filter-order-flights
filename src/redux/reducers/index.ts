import filteringOptionsReducer from "./filteringOptionsReducer"
import filtersReducer from "./filtersReducer"
import itinerariesReducer from "./itinerariesReducer"

export { filteringOptionsReducer, filtersReducer, itinerariesReducer }
export * from "./rootReducer"
