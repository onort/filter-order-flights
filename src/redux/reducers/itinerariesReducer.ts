import { Reducer } from "redux"

import { DataActionTypes, FetchDataAction } from "../actions"
import { Itinerary } from "../../types"

const itinerariesReducer: Reducer<Itinerary[], FetchDataAction> = (
  state = [],
  action: FetchDataAction
) => {
  switch (action.type) {
    case DataActionTypes.FETCH_DATA:
      return action.data.itineraries
    default:
      return state
  }
}

export default itinerariesReducer
