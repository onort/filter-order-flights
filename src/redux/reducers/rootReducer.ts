import { combineReducers, Reducer } from "redux"

import { filteringOptionsReducer, itinerariesReducer } from "./"

export interface ApplicationState {
  readonly filteringOptions: object
  readonly itineraries: any[]
}

export const rootReducer: Reducer = combineReducers({
  filteringOptions: filteringOptionsReducer,
  itineraries: itinerariesReducer
})

export default rootReducer
