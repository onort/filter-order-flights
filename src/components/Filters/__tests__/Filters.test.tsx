import React from "react"
import { shallow, ShallowWrapper } from "enzyme"

import { Filters } from "../Filters"
import { initialState as filtersInitialState } from "../../../redux/reducers/filtersReducer"
import { initialState as filteringOptionsInitialState } from "../../../redux/reducers/filteringOptionsReducer"

describe("<Filters />", () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    const allAirlines = jest.fn()
    const allClasses = jest.fn()
    const allStops = jest.fn()
    const setDuration = jest.fn()
    const setPrice = jest.fn()
    const toggleAirline = jest.fn()
    const toggleClass = jest.fn()
    const toggleStop = jest.fn()

    wrapper = shallow(
      <Filters
        allAirlines={allAirlines}
        allClasses={allClasses}
        allStops={allStops}
        setDuration={setDuration}
        setPrice={setPrice}
        toggleAirline={toggleAirline}
        toggleClass={toggleClass}
        toggleStop={toggleStop}
        filteringOptions={filteringOptionsInitialState}
        filters={filtersInitialState}
      />
    )
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
