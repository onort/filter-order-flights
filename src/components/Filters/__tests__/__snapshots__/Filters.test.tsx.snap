// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    allAirlines={[MockFunction]}
    allClasses={[MockFunction]}
    allStops={[MockFunction]}
    filteringOptions={
      Object {
        "airlineCodes": Array [],
        "airlines": Array [],
        "durationRange": Object {
          "max": 0,
          "min": 0,
        },
        "priceRange": Object {
          "max": 0,
          "min": 0,
        },
        "stops": Object {
          "direct": 0,
          "multipleStops": 0,
          "singleStop": 0,
        },
      }
    }
    filters={
      Object {
        "airlines": Array [],
        "classes": Array [
          "economy",
          "business",
          "corporate",
        ],
        "maxDuration": 0,
        "maxPrice": 0,
        "stops": Array [
          "direct",
          "singleStop",
          "multipleStops",
        ],
      }
    }
    setDuration={[MockFunction]}
    setPrice={[MockFunction]}
    toggleAirline={[MockFunction]}
    toggleClass={[MockFunction]}
    toggleStop={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card>
        <FilterSection
          canSelectAll={true}
          onSelectAll={[Function]}
          title="Aktarma"
        />
        <FilterSection
          canSelectAll={true}
          onSelectAll={[Function]}
          title="Sınıf"
        >
          <CheckBox
            checked={true}
            label="Ekonomi"
            onChange={[Function]}
          />
          <CheckBox
            checked={true}
            label="Business"
            onChange={[Function]}
          />
          <CheckBox
            checked={true}
            label="Kurumsal"
            onChange={[Function]}
          />
        </FilterSection>
        <FilterSection
          canSelectAll={false}
          infoText="0 TL"
          title="Max Fiyat"
        >
          <Slider
            currentValue={0}
            max={0}
            min={0}
            onChange={[Function]}
          />
        </FilterSection>
        <FilterSection
          canSelectAll={false}
          infoText=""
          title="Max Uçuş Süresi"
        >
          <Slider
            currentValue={0}
            max={0}
            min={0}
            onChange={[Function]}
          />
        </FilterSection>
        <FilterSection
          canSelectAll={true}
          onSelectAll={[Function]}
          title="Havayolu"
        />
      </Card>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FilterSection
            canSelectAll={true}
            onSelectAll={[Function]}
            title="Aktarma"
          />,
          <FilterSection
            canSelectAll={true}
            onSelectAll={[Function]}
            title="Sınıf"
          >
            <CheckBox
              checked={true}
              label="Ekonomi"
              onChange={[Function]}
            />
            <CheckBox
              checked={true}
              label="Business"
              onChange={[Function]}
            />
            <CheckBox
              checked={true}
              label="Kurumsal"
              onChange={[Function]}
            />
          </FilterSection>,
          <FilterSection
            canSelectAll={false}
            infoText="0 TL"
            title="Max Fiyat"
          >
            <Slider
              currentValue={0}
              max={0}
              min={0}
              onChange={[Function]}
            />
          </FilterSection>,
          <FilterSection
            canSelectAll={false}
            infoText=""
            title="Max Uçuş Süresi"
          >
            <Slider
              currentValue={0}
              max={0}
              min={0}
              onChange={[Function]}
            />
          </FilterSection>,
          <FilterSection
            canSelectAll={true}
            onSelectAll={[Function]}
            title="Havayolu"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "canSelectAll": true,
            "children": Array [
              false,
              false,
              false,
            ],
            "onSelectAll": [Function],
            "title": "Aktarma",
          },
          "ref": null,
          "rendered": Array [
            false,
            false,
            false,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "canSelectAll": true,
            "children": Array [
              <CheckBox
                checked={true}
                label="Ekonomi"
                onChange={[Function]}
              />,
              <CheckBox
                checked={true}
                label="Business"
                onChange={[Function]}
              />,
              <CheckBox
                checked={true}
                label="Kurumsal"
                onChange={[Function]}
              />,
            ],
            "onSelectAll": [Function],
            "title": "Sınıf",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "label": "Ekonomi",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "label": "Business",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "label": "Kurumsal",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "canSelectAll": false,
            "children": <Slider
              currentValue={0}
              max={0}
              min={0}
              onChange={[Function]}
            />,
            "infoText": "0 TL",
            "title": "Max Fiyat",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentValue": 0,
              "max": 0,
              "min": 0,
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "canSelectAll": false,
            "children": <Slider
              currentValue={0}
              max={0}
              min={0}
              onChange={[Function]}
            />,
            "infoText": "",
            "title": "Max Uçuş Süresi",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentValue": 0,
              "max": 0,
              "min": 0,
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "canSelectAll": true,
            "children": Array [],
            "onSelectAll": [Function],
            "title": "Havayolu",
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card>
          <FilterSection
            canSelectAll={true}
            onSelectAll={[Function]}
            title="Aktarma"
          />
          <FilterSection
            canSelectAll={true}
            onSelectAll={[Function]}
            title="Sınıf"
          >
            <CheckBox
              checked={true}
              label="Ekonomi"
              onChange={[Function]}
            />
            <CheckBox
              checked={true}
              label="Business"
              onChange={[Function]}
            />
            <CheckBox
              checked={true}
              label="Kurumsal"
              onChange={[Function]}
            />
          </FilterSection>
          <FilterSection
            canSelectAll={false}
            infoText="0 TL"
            title="Max Fiyat"
          >
            <Slider
              currentValue={0}
              max={0}
              min={0}
              onChange={[Function]}
            />
          </FilterSection>
          <FilterSection
            canSelectAll={false}
            infoText=""
            title="Max Uçuş Süresi"
          >
            <Slider
              currentValue={0}
              max={0}
              min={0}
              onChange={[Function]}
            />
          </FilterSection>
          <FilterSection
            canSelectAll={true}
            onSelectAll={[Function]}
            title="Havayolu"
          />
        </Card>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <FilterSection
              canSelectAll={true}
              onSelectAll={[Function]}
              title="Aktarma"
            />,
            <FilterSection
              canSelectAll={true}
              onSelectAll={[Function]}
              title="Sınıf"
            >
              <CheckBox
                checked={true}
                label="Ekonomi"
                onChange={[Function]}
              />
              <CheckBox
                checked={true}
                label="Business"
                onChange={[Function]}
              />
              <CheckBox
                checked={true}
                label="Kurumsal"
                onChange={[Function]}
              />
            </FilterSection>,
            <FilterSection
              canSelectAll={false}
              infoText="0 TL"
              title="Max Fiyat"
            >
              <Slider
                currentValue={0}
                max={0}
                min={0}
                onChange={[Function]}
              />
            </FilterSection>,
            <FilterSection
              canSelectAll={false}
              infoText=""
              title="Max Uçuş Süresi"
            >
              <Slider
                currentValue={0}
                max={0}
                min={0}
                onChange={[Function]}
              />
            </FilterSection>,
            <FilterSection
              canSelectAll={true}
              onSelectAll={[Function]}
              title="Havayolu"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "canSelectAll": true,
              "children": Array [
                false,
                false,
                false,
              ],
              "onSelectAll": [Function],
              "title": "Aktarma",
            },
            "ref": null,
            "rendered": Array [
              false,
              false,
              false,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "canSelectAll": true,
              "children": Array [
                <CheckBox
                  checked={true}
                  label="Ekonomi"
                  onChange={[Function]}
                />,
                <CheckBox
                  checked={true}
                  label="Business"
                  onChange={[Function]}
                />,
                <CheckBox
                  checked={true}
                  label="Kurumsal"
                  onChange={[Function]}
                />,
              ],
              "onSelectAll": [Function],
              "title": "Sınıf",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "label": "Ekonomi",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "label": "Business",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "label": "Kurumsal",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "canSelectAll": false,
              "children": <Slider
                currentValue={0}
                max={0}
                min={0}
                onChange={[Function]}
              />,
              "infoText": "0 TL",
              "title": "Max Fiyat",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currentValue": 0,
                "max": 0,
                "min": 0,
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "canSelectAll": false,
              "children": <Slider
                currentValue={0}
                max={0}
                min={0}
                onChange={[Function]}
              />,
              "infoText": "",
              "title": "Max Uçuş Süresi",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currentValue": 0,
                "max": 0,
                "min": 0,
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "canSelectAll": true,
              "children": Array [],
              "onSelectAll": [Function],
              "title": "Havayolu",
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<Filters /> should match snapshot 2`] = `"<aside class=\\"container\\"><div class=\\"container\\"><div class=\\"container\\"><div class=\\"header\\"><span class=\\"title\\">Aktarma</span><span class=\\"selectAll\\">Tümünü Seç</span></div><div class=\\"body\\"></div></div><div class=\\"container\\"><div class=\\"header\\"><span class=\\"title\\">Sınıf</span><span class=\\"selectAll\\">Tümünü Seç</span></div><div class=\\"body\\"><label class=\\"container\\">Ekonomi<input type=\\"checkbox\\" class=\\"checkbox\\" checked=\\"\\"/><span class=\\"checkmark checked\\"></span></label><label class=\\"container\\">Business<input type=\\"checkbox\\" class=\\"checkbox\\" checked=\\"\\"/><span class=\\"checkmark checked\\"></span></label><label class=\\"container\\">Kurumsal<input type=\\"checkbox\\" class=\\"checkbox\\" checked=\\"\\"/><span class=\\"checkmark checked\\"></span></label></div></div><div class=\\"container\\"><div class=\\"header\\"><span class=\\"title\\">Max Fiyat</span><span class=\\"infoText\\">0 TL</span></div><div class=\\"body\\"><div class=\\"container\\"><div class=\\"sliderTrack\\" style=\\"width:NaN%\\"></div><input type=\\"range\\" min=\\"0\\" max=\\"0\\" value=\\"0\\" class=\\"slider\\"/></div></div></div><div class=\\"container\\"><div class=\\"header\\"><span class=\\"title\\">Max Uçuş Süresi</span></div><div class=\\"body\\"><div class=\\"container\\"><div class=\\"sliderTrack\\" style=\\"width:NaN%\\"></div><input type=\\"range\\" min=\\"0\\" max=\\"0\\" value=\\"0\\" class=\\"slider\\"/></div></div></div><div class=\\"container\\"><div class=\\"header\\"><span class=\\"title\\">Havayolu</span><span class=\\"selectAll\\">Tümünü Seç</span></div><div class=\\"body\\"></div></div></div></aside>"`;
